device/cuda/src/seeding/doublet_counting.cu

```diff
@@ -12,6 +12,12 @@
 namespace traccc {
 namespace cuda {
 
+__global__ void set_zero_kernel(doublet_counter_container_view dcc_view) {
+    device_doublet_counter_container dcc_device(
+        {dcc_view.headers, dcc_view.items});
+    dcc_device.get_headers().at(threadIdx.x).zeros();
```

Comment:
You should add a

```c++
if (threadIdx.x >= dcc_device.get_headers().size()) {
  return;
}
```

check. Or better, the full

```c++
const std::size_t i = blockIdx.x * blockSize.x + threadIdx.x;
if (i >= >= dcc_device.get_headers().size()) {
  return;
}
```

, as you'll see in my next comment...

Reply:
Indeed. Resolved.